### PR TITLE
Return unsupported token error fee route

### PR DESCRIPTION
**Pull request by fleupold**
Following #550 this PR now makes it so we return this error for unsupported tokens

### Test Plan
Added unit test + starting orderbook locally with `--unsupported-tokens 0xfad45e47083e4607302aa43c65fb3106f1cd7607` and on 

```
http://localhost:8080/api/v1/fee?sellToken=0xfad45e47083e4607302aa43c65fb3106f1cd7607&buyToken=0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d&amount=1000000&kind=buy
```

Getting

> {"errorType":"UnsupportedToken","description":"Token address 0xfad45e47083e4607302aa43c65fb3106f1cd7607"}

### CODE REVIEW

File: orderbook/src/fee.rs [hunk 1]
```diff
@@ -97,6 +100,19 @@ impl MinFeeCalculator {
...
+        }
+
+        if buy_token
+            .map(|t| self.unsupported_tokens.contains(&t))
+            .unwrap_or_default()
+        {
+            return Err(MinFeeCalculationError::UnsupportedToken(
+                buy_token.expect("Must exist"),
+            ));
+        }
```
Comment (vkgnosis) [inline orderbook/src/fee.rs:114]:
```suggestion
        if let Some(token) = buy_token {
            if self.unsupported_tokens.contains(&token) {
                return Err(MinFeeCalculationError::UnsupportedToken(token));
            }
        }
```

**Review by vkgnosis — APPROVED** [review]

File: orderbook/src/fee.rs (hunk 1 above)
Comment (fleupold) [reply orderbook/src/fee.rs:114]: Right, I felt the upper was easier to merge with #542

File: orderbook/src/api/get_fee_info.rs [hunk 2]
```diff
@@ -50,7 +50,7 @@ pub fn get_fee_info_response(
             StatusCode::NOT_FOUND,
         )),
         Err(MinFeeCalculationError::UnsupportedToken(token)) => Ok(reply::with_status(
-            super::error("UnsupportedToken", format!("Token address {}", token)),
+            super::error("UnsupportedToken", format!("Token address {:?}", token)),
```
Comment (bh2smith) [inline orderbook/src/api/get_fee_info.rs:53]: Whoops! My mistake. Does this mean the token address wasn't being displayed before?

**Review by bh2smith — APPROVED** [review]

File: orderbook/src/api/get_fee_info.rs (hunk 2 above)
Comment (fleupold) [reply orderbook/src/api/get_fee_info.rs:53]: Actually this was my mistake in #550

**Pull request merged**